src/main/java/org/ehrbase/fhirbridge/ehr/converter/TherapyCompositionConverter.java

```diff
@@ -0,0 +1,264 @@
+import org.ehrbase.fhirbridge.ehr.opt.geccoprozedurcomposition.definition.GeccoProzedurKategorieElement;
+import org.ehrbase.fhirbridge.ehr.opt.geccoprozedurcomposition.definition.ProzedurAction;
+import org.ehrbase.fhirbridge.ehr.opt.geccoprozedurcomposition.definition.UnbekannteProzedurEvaluation;
+import org.ehrbase.fhirbridge.ehr.opt.geccoprozedurcomposition.definition.MedizingeraetCluster;
+import org.ehrbase.fhirbridge.ehr.opt.geccoprozedurcomposition.definition.NichtDurchgefuehrteProzedurEvaluation;
+import org.ehrbase.fhirbridge.ehr.opt.geccoprozedurcomposition.definition.CareflowStepDefiningCode;
+import org.ehrbase.fhirbridge.ehr.opt.geccoprozedurcomposition.definition.CurrentStateDefiningCode;
+import org.hl7.fhir.exceptions.FHIRException;
+import org.hl7.fhir.r4.model.Extension;
+import org.hl7.fhir.r4.model.Procedure;
+import org.hl7.fhir.r4.model.DateTimeType;
+import org.hl7.fhir.r4.model.Coding;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.GregorianCalendar;
+import java.util.HashMap;
+import java.util.Map;
+
+
+public class TherapyCompositionConverter implements CompositionConverter<GECCOProzedurComposition, Procedure> {
+
+    private static final Logger LOG = LoggerFactory.getLogger(TherapyCompositionConverter.class);
+
+    private static final Map<String, KategorieDefiningCode> kategorieMap = new HashMap<>();
+    private static final Map<String, NameDerProzedurDefiningCode> nameDerProzedurMap = new HashMap<>();
+    private static final Map<String, KoerperstelleDefiningCode> koerperstelleMap = new HashMap<>();
+    private static final Map<String, GeraetenameDefiningCode> geraetenameMap = new HashMap<>();
+
+    private static String SNOMED_SYSTEM = "http://snomed.info/sct";
+
+
+    static {
+        for (KategorieDefiningCode kategorie : KategorieDefiningCode.values()) {
+            kategorieMap.put(kategorie.getCode(), kategorie);
+        }
+
+        for (NameDerProzedurDefiningCode nameDerProzedurDefiningCode : NameDerProzedurDefiningCode.values()) {
+            nameDerProzedurMap.put(nameDerProzedurDefiningCode.getCode(), nameDerProzedurDefiningCode);
+        }
+
+        for (KoerperstelleDefiningCode koerperstelleDefiningCode : KoerperstelleDefiningCode.values()) {
+            koerperstelleMap.put(koerperstelleDefiningCode.getCode(), koerperstelleDefiningCode);
+        }
+
+        for (GeraetenameDefiningCode geraetenameDefiningCode : GeraetenameDefiningCode.values()) {
+            geraetenameMap.put(geraetenameDefiningCode.getCode(), geraetenameDefiningCode);
+        }
+    }
+
+
+    @Override
+    public Procedure fromComposition(GECCOProzedurComposition composition) throws CompositionConversionException {
+        // TODO: Implement
+        return null;
+    }
+
+    @Override
+    public GECCOProzedurComposition toComposition(Procedure procedure) throws CompositionConversionException {
```

Comment:
Method is way to long, please seperate different functionality in different methods. Aim for a maximum of 30 rows.

Reply:
Fixed